src/python/WMCore/MicroService/Unified/MSMonitor.py

```diff
@@ -206,6 +216,47 @@ def _getTransferstatus(self, dataset, requestList):
             return 0
         return sum(completion) / len(completion)
 
+    def _getRucioTransferstatus(self, requestList):
+        """
+        Fetch the transfer request status from PhEDEx
+        :param dataset: dataset name string
+        :param requestList: list of request IDs
+        :return: the overall transfers percent completion
+
+        The Rucio getRule API returns data in the form of:
+            {u'account': u'transfer_ops',
+             u'grouping': u'ALL',
+             u'id': u'40cbe787a42b4f6e991611f6fac3bb11',
+             u'locked': True,
+             u'locks_ok_cnt': 8,
+             u'locks_replicating_cnt': 0,
+             u'locks_stuck_cnt': 0,
+             u'meta': None,
+             etc etc
+            """
+        completion = []
+        for reqId in requestList:
+            # if we query by dataset and the subscription was at block level,
+            # we get an empty response. So always wildcard the block parameter
+            data = self.rucio.getRule(reqId)
+            if not data:
+                # failed to retrieve data for this rule id,
+                # TODO: should we raise an exception and work on it in the next cycle?
+                completion.append(0)
+                continue
+
+            if data['state'] == "OK":
```

Comment:
@ericvaandering Eric, could you please confirm whether this is all we need to track the rucio rule progress?
In short:
state=OK means the rule is 100% completed
otherwise locks_ok / sum(all the other 3 locks) gives me the progress of this yet incomplete rule.

Does it cover it properly?
Thanks!

Reply:
Yes. Still looking. Can you change reqID to ruleID? Requests are another thing in Rucio and I'd rather not have anyone get confused. I presume request list stays as is since it can be PhEDEx or Rucio request.

Reply:
Sure, I'm updating this. Eventually I will stop mixing up those words with different meanings ;)